### ISSUE TITLE
gobject-introspection upgrade fails: bug fixed in latest version but fix not back-ported to the version homebrew is trying to install

### ISSUE THREAD

**Issue by maspotts** [Homebrew/homebrew-core#93211]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 3.3.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: 385892f3d27dcab0f5a1cb47c3927d9d65ed007c
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c34e36ed9feb170d4b10ed72d5bb4cb6f83d2a5f
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/brew
HOMEBREW_REPOSITORY: /opt/brew/Homebrew
HOMEBREW_CELLAR: /opt/brew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: vi
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_ANALYTICS: set
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 13.0.0 build 1300
Git: 2.34.1 => /opt/brew/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A


Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/mike/.pyenv/shims/python2-config
  /Users/mike/.pyenv/shims/python2.7-config
  /Users/mike/.pyenv/shims/python3.9-config
  /Users/mike/.pyenv/shims/python-config
  /Users/mike/.pyenv/shims/python3-config

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  ffmpeg
  gobject-introspection

Warning: Your Homebrew's prefix is not /usr/local.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/usr/local).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.


NOTE: I cannot fix the unlinked kegs: those are unlinked because gobject-introspection will not upgrade, and ffmpeg requires it.

https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/415
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

brew upgrade
  -> trying to upgrade ffmpeg
  -> has a dependency on gobject-introspection
  -> gobject-introspection upgrade fails due to bug that is fixed in latest version, but the fix has not been back-ported to the version that homebrew is trying to install.

See this gobject-introspection issue citing this problem:

https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/415

### What happened (include all command output)?

brew upgrade gobject-introspection
==> Upgrading 1 outdated package:
gobject-introspection 1.70.0_2 -> 1.70.0_3
Warning: Building gobject-introspection from source as the bottle needs:
- HOMEBREW_CELLAR: /usr/local/Cellar (yours is /opt/brew/Cellar)
- HOMEBREW_PREFIX: /usr/local (yours is /opt/brew)
==> Downloading https://gitlab.gnome.org/tschoonj/gobject-introspection/-/commit/a7be304478b25271166cd92d110f251a8742d16b.diff
Already downloaded: /Users/mike/Library/Caches/Homebrew/downloads/8b05b2a9de1da2191fe40b8fb80652d77efaa426382912d3d850e303bcf08d38--a7be304478b25271166cd92d110f251a8742d16b.diff
==> Downloading https://gitlab.gnome.org/rokm/gobject-introspection/-/commit/56df7b0f007fe260b2bd26ef9cc331ad73022700.diff
Already downloaded: /Users/mike/Library/Caches/Homebrew/downloads/cf90f085e421f2f49eef9de6383e9ad30c0cddae0e837f541a20d2c0b7b4404d--56df7b0f007fe260b2bd26ef9cc331ad73022700.diff
==> Cloning https://gist.github.com/7a0023656ccfe309337a.git
Updating /Users/mike/Library/Caches/Homebrew/gobject-introspection--tutorial--git
==> Checking out revision 499ac89f8a9ad17d250e907f74912159ea216416
HEAD is now at 499ac89
HEAD is now at 499ac89
==> Downloading https://download.gnome.org/sources/gobject-introspection/1.70/gobject-introspection-1.70.0.tar.xz
Already downloaded: /Users/mike/Library/Caches/Homebrew/downloads/521d3ce8b76d8614697dc2dcf53d58c641569d12aed515db213ab68208976101--gobject-introspection-1.70.0.tar.xz
==> Upgrading gobject-introspection
  1.70.0_2 -> 1.70.0_3

==> Patching
==> Applying a7be304478b25271166cd92d110f251a8742d16b.diff
patching file girepository/gitypelib.c
Hunk #1 succeeded at 2261 (offset 2 lines).
patching file meson.build
Hunk #1 succeeded at 122 (offset -2 lines).
patching file meson_options.txt
==> Applying 56df7b0f007fe260b2bd26ef9cc331ad73022700.diff
patching file girepository/gitypelib.c
Hunk #1 succeeded at 2278 with fuzz 2 (offset 16 lines).
==> meson --prefix=/opt/brew/Cellar/gobject-introspection/1.70.0_3 --libdir=/opt/brew/Cellar/gobject-introspection/1.70.0_3/lib --buildtype=release --wrap-mode=nofallback -Dpython=/opt/brew/opt/python@3.9/bin/py
Last 15 lines from /Users/mike/Library/Logs/Homebrew/gobject-introspection/01.meson:
Configuring message.py using configuration
Configuring msvccompiler.py using configuration
Configuring pkgconfig.py using configuration
Configuring shlibs.py using configuration
Configuring scannermain.py using configuration
Configuring sectionparser.py using configuration
Configuring sourcescanner.py using configuration
Configuring testcodegen.py using configuration
Configuring transformer.py using configuration
Configuring utils.py using configuration
Configuring xmlwriter.py using configuration

../giscanner/meson.build:58:0: ERROR: File doctemplates does not exist.

A full log can be found at /private/tmp/gobject-introspection-20220115-94169-1pmvzv0/gobject-introspection-1.70.0/build/meson-logs/meson-log.txt

Do not report this issue to Homebrew/brew or Homebrew/core!

### What did you expect to happen?

successful upgrade of gobject-introspection and hence upgrade of all remaining formulae.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew upgrade gobject-introspection
```

**Comment by carlocab** [comment]
Duplicate of #92992

**Comment by carlocab** [comment]
Fixed in 3e50746ab6135c2b802d378a5164e18667700e42.
